platform/ui/src/components/ViewportOverlay/index.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import ConfigPoint from 'config-point'
+import ListContent from './ListContent';
+import PropTypes from 'prop-types';
+import itemGenerator from './itemGenerator';
+
+/**
+ * Generates a viewport overlay given a configuration element.
+ * The element typically comes from config-point, and contains
+ * things like the topleft/topRight/bottomleft/bottomRight definitions.
+ * @param {*} config is a configuration object that defines four lists of elements,
+ * one topLeft, topRight, bottomLeft, bottomRight contents.
+ * @returns React Viewport overlay function configured to use config for display
+ */
+const generateFromConfig = (config) => {
```

Comment:
I understand the premise here, but I have the following comments:

- `@ohif/ui` is supposed to be a presentational component library (although it does admitly fail this on some components, which need to be improved). By using the `ConfigPoint` to generate components, we might be opening the door to effectively making them stateful, although I don't think that is quite done in this PR, but the API surface sure is a lot larger than before.
- I fail to see why we can't just achieve this logic with Higher Order Components (HOC's) instead, and leave the UI library as plain JSX driven by props:
  - Have a `configurationService` at the top level passed into `extensions` and `modes`  via the `extension`'s `preRegistration` hook, and the `mode`'s `onModeEnter`.
  - Improve the UI components in platform UI to take configuration as one of its props, which is used to populate the UI element with E.g. fields, other child components, etc.
  - The extension can create a configuration object by merging the extensions object with the set configuration.
  - The extension can then create a HOC which wraps the UI component, and passes this configuration in as props (e.g. similar to how the  `wrappedMeasurementPanel` component in `extensions/default/src/getPanelModule.js` passes services to the panel module).
  
For all the examples in this PR, right now the extension would do the heavy lifting of configuring its components with HOCs (since the configs used here are hardcoded in the extensiosn), and a configurationService wouldn't actually be needed at all. But I understand how configuration from the app level in future would be important also.

These are all high level thoughts and discussions, its very possible I missed something fundemental which makes this approach better.

Reply:
> Obviously I didn't quite hit the mark the first time

FWIW I'm not suggesting that in any of these comments actually, I'm not sure of the answer, just thinking out loud here so we all come to the best place for the OSS lib :) . 

> but I'm going to try to update the PR a bit to show exactly this idea.
Sounds good, there may be entire swathes of problems that this approach solves that I'm just not thinking of yet.